parametric/apps/golang/main.go

```diff
@@ -44,11 +44,19 @@ func newServer() *apmClientServer {
 		spans:     make(map[uint64]tracer.Span),
 		otelSpans: make(map[uint64]ot_api.Span),
 	}
+	s.tp = ot.NewTracerProvider()
```

Comment:
Doesn't this also happen in `StartTracer`? (Is it intentional to have the otel trace provider set in two different places for different kinds of tests?)

Reply:
I have removed StartTracer in this iteration